Cystic fibrosis respiratory microbiome data were retrieved from studies with publicly available sequencing data (1-18). The pre-ETI samples were downloaded from the Dartmouth RESPIRE Database (https://respire.dartmouth.edu/), while the post-ETI samples were downloaded from NCBI BioProject using the Google Cloud service BigQuery.

Prevalence vs. Relative Abundance:
Prevalence and relative abundance of the different genera were determined using a modified version of an established source code (19). Relative abundances were calculated by normalizing each sample’s genus-level read counts to its total reads. Prevalence was computed as the fraction of samples in which each genus was detected, and mean relative abundance was averaged across samples. Only genera present in more than 20% of samples were retained, and mean abundances were log-transformed. The most abundant genera were selected for plotting in each dataset. Plots were generated using ggplot2 (20), labelled using ggrepel (21). The dplyr package (22) was used for data manipulation.

Network Analysis:
Microbial correlation network analysis was performed using the igraph (23-25), Hmisc (26), and corrplot (27) packages to explore co-occurrence patterns among the most abundant genera across multiple sample types and clinical groups before and after ETI treatment. Pairwise Spearman correlations were computed using relative abundance data. Spearman correlations that had an |r| > 0.5 and were statistically significant (p < 0.05) were retained and visualized via heatmaps. Microbial co-occurrence networks were constructed to be undirected and weighted, with nodes representing genera and edges denoting positive or negative correlations. Node sizes were scaled by degree centrality, and edge widths reflected correlation strength. Network metrics such as degree and edge density were calculated, and community detection was performed using the Louvain algorithm (28) to identify clusters of co-associated genera.

References:
1. Flight WG, Smith A, Paisey C, Marchesi JR, Bull MJ, Norville PJ, Mutton KJ, Webb AK, Bright-Thomas RJ, Jones AM, Mahenthiralingam E. 2015. Rapid detection of emerging pathogens and loss of microbial diversity associated with severe lung disease in cystic fibrosis. J Clin Microbiol 53:2022–2029
2. Muhlebach MS, Zorn BT, Esther CR, Hatch JE, Murray CP, Turkovic L, Ranganathan SC, Boucher RC, Stick SM, Wolfgang MC. 2018. Initial acquisition and succession of the cystic fibrosis lung microbiome is associated with disease progression in infants and preschool children. PLoS Pathog 14:e1006798
3. Kramná L, Dřevínek P, Lin J, Kulich M, Cinek O. 2018. Changes in the lung bacteriome in relation to antipseudomonal therapy in children with cystic fibrosis. Folia Microbiol (Praha) 63:237–248
4. Whelan FJ, Heirali AA, Rossi L, Rabin HR, Parkins MD, Surette MG. 2017. Longitudinal sampling of the lung microbiota in individuals with cystic fibrosis. PLOS ONE 12:e0172811
5. Hahn A, Burrell A, Fanous H, Chaney H, Sami I, Perez GF, Koumbourlis AC, Freishtat RJ, Crandall KA. 2018. Antibiotic multidrug resistance in the cystic fibrosis airway microbiome is associated with decreased diversity. Heliyon 4:e00795
6. Quinn RA, Adem S, Mills RH, Comstock W, DeRight Goldasich L, Humphrey G, Aksenov AA, Melnik AV, da Silva R, Ackermann G, Bandeira N, Gonzalez DJ, Conrad D, O’Donoghue AJ, Knight R, Dorrestein PC. 2019. Neutrophilic proteolysis in the cystic fibrosis lung correlates with a pathogenic microbiome. Microbiome 7:23
7. Linnane B, Walsh AM, Walsh CJ, Crispie F, O’Sullivan O, Cotter PD, McDermott M, Renwick J, McNally P. 2021. The lung microbiome in young children with cystic fibrosis: a prospective cohort study. Microorganisms 9:492
8. Cuthbertson L, Walker AW, Oliver AE, Rogers GB, Rivett DW, Hampton TH, Ashare A, Elborn JS, De Soyza A, Carroll MP, Hoffman LR, Lanyon C, Moskowitz SM, O’Toole GA, Parkhill J, Planet PJ, Teneback CC, Tunney MM, Zuckerman JB, Bruce KD, van der Gast CJ. 2020. Lung function and microbiota diversity in cystic fibrosis. Microbiome 8:45
9. Harris JK, Wagner BD, Robertson CE, Stevens MJ, Lingard C, Borowitz D, Leung DH, Heltshe SL, Ramsey BW, Zemanick ET. 2023. Upper airway microbiota development in infants with cystic fibrosis diagnosed by newborn screen. J Cyst Fibros 22:644–651
10. Batson BD, Zorn BT, Radicioni G, Livengood SS, Kumagai T, Dang H, Ceppe A, Clapp PW, Tunney M, Elborn JS, McElvaney NG, Muhlebach MS, Boucher RC, Tiemeyer M, Wolfgang MC, Kesimer M. 2022. Cystic fibrosis airway mucus hyperconcentration produces a vicious cycle of mucin, pathogen, and inflammatory interactions that promotes disease persistence. Am J Respir Cell Mol Biol 67:253–265
11. Garcia-Nuñez M, Garcia-Gonzalez M, Pomares X, Montón C, Millares L, Quero S, Prina E, Asensio O, Bosque M, Capilla S, Cuevas O, Monsó E. 2020. The respiratory microbiome in cystic fibrosis: compartment patterns and clinical relationships in early stage disease. Front Microbiol 11:1463
12. Zhao CY, Hao Y, Wang Y, Varga JJ, Stecenko AA, Goldberg JB, Brown SP. 2021. Microbiome data enhances predictive models of lung function in people with cystic fibrosis. J Infect Dis 223:S246–S256
13. Sarwar PF. 2022. Prevotella phylogeny: genomic and molecular insights into the role of the human commensal Prevotella in cystic fibrosis. ProQuest Dissertations and Theses. Ph.D. University of Pennsylvania, United States -- Pennsylvania
14. Sosinski LM, H CM, Neugebauer KA, Ghuneim L-AJ, Guzior DV, Castillo-Bahena A, Mielke J, Thomas R, McClelland M, Conrad D, Quinn RA. 2022. A restructuring of microbiome niche space is associated with Elexacaftor-Tezacaftor-Ivacaftor therapy in the cystic fibrosis lung. J Cyst Fibros 21:996–1005
15. Hilliam Y, Armbruster CR, Rapsinski GJ, Marshall CW, Moore J, Koirala J, Krainz L, Gaston JR, Cooper VS, Lee SE, Bomberger JM. 2024. Cystic fibrosis pathogens persist in the upper respiratory tract following initiation of elexacaftor/tezacaftor/ivacaftor therapy. Microbiol Spectr 12:e0078724
16. Hogan DA, Willger SD, Dolben EL, Hampton TH, Stanton BA, Morrison HG, Sogin ML, Czum J, Ashare A. 2016. Analysis of lung microbiota in bronchoalveolar lavage, protected brush and sputum samples from subjects with mild-to-moderate cystic fibrosis lung disease. PLoS One 11:e0149998
17. Nichols DP, Morgan SJ, Skalland M, Vo AT, Van Dalfsen JM, Singh SB, Ni W, Hoffman LR, McGeer K, Heltshe SL, Clancy JP, Rowe SM, Jorth P, Singh PK, PROMISE-Micro Study Group. 2023. Pharmacologic improvement of CFTR function rapidly decreases sputum pathogen density, but lung infections generally persist. J Clin Invest 133:e167957
18. Zemke AC, Hilliam Y, Stapleton AL, Kimple AJ, Goralski JL, Shaffer AD, Pilewski JM, Senior BA, Lee SE, Cooper VS. 2024. Elexacaftor-tezacaftor-ivacaftor decreases pseudomonas abundance in the sinonasal microbiome in cystic fibrosis. Int Forum Allergy Rhinol 14:928–938
19. Jean-Pierre F, Hampton TH, Schultz D, Hogan DA, Groleau M-C, Déziel E, O’Toole GA. 2023. Community composition shapes microbial-specific phenotypes in a cystic fibrosis polymicrobial model system. eLife 12:e81604
20. Wickham H. 2016. ggplot2: elegant graphics for data analysis. Springer-Verlag New York. https://ggplot2.tidyverse.org
21. Slowikowski K. 2024. ggrepel: automatically position non-overlapping text labels with “ggplot2.” https://ggrepel.slowkow.com/, https://github.com/slowkow/ggrepel
22. Wickham H, François R, Henry L, Müller K, Vaughan D. 2025. dplyr: a grammar of data manipulation. https://dplyr.tidyverse.org
23. Csárdi G, Nepusz T. 2006. The igraph software package for complex network research. InterJournal Complex Systems:1695.
24. Antonov M, Csárdi G, Horvát S, Müller K, Nepusz T, Noom D, Salmon M, Traag V, Welles BF, Zanini F. 2023. igraph enables fast and robust network analysis across programming languages (1). arXiv https://doi.org/10.48550/ARXIV.2311.10260.
25. Csárdi G, Nepusz T, Müller K, Horvát S, Traag V, Zanini F, Noom D. 2025. igraph for R: R interface of the igraph library for graph theory and network analysis (v2.2.0). Zenodo
26. Harrell Jr FE. 2025. Hmisc: Harrell Miscellaneous. https://github.com/harrelfe/hmisc.
27. Wei T, Simko V. 2024. R package “corrplot”: visualization of a correlation matrix. https://github.com/taiyun/corrplot
28. Blondel VD, Guillaume J-L, Lambiotte R, Lefebvre E. 2008. Fast unfolding of communities in large networks. Journal of Statistical Mechanics: Theory and Experiment 2008:P10008
